eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriterTest.java

```diff
@@ -234,17 +249,54 @@ public void repeatedClusteredVariants() throws Exception {
+                SUBMITTED_VARIANT_ACCESSION, hashingFunctionSubmitted.apply(submittedVariant2), submittedVariant2, 1);
+        DbsnpVariantsWrapper wrapper2 = buildSimpleWrapper(Collections.singletonList(submittedVariantEntity2));
+
+        dbsnpVariantsWriter.write(Arrays.asList(wrapper1, wrapper2));
+        assertClusteredVariantStored(wrapper1);
+    }
+
+    @Test
+    public void repeatedClusteredVariantsDeclustered() throws Exception {
+        boolean allelesMatch = false;
+        SubmittedVariant submittedVariant1 = new SubmittedVariant("assembly", TAXONOMY_2, "project", "contig", START,
+                                                                  "reference", "alternate", CLUSTERED_VARIANT_ACCESSION,
+                                                                  DEFAULT_SUPPORTED_BY_EVIDENCE, DEFAULT_ASSEMBLY_MATCH,
+                                                                  allelesMatch, DEFAULT_VALIDATED);
+        DbsnpSubmittedVariantEntity submittedVariantEntity1 = new DbsnpSubmittedVariantEntity(
+                SUBMITTED_VARIANT_ACCESSION, hashingFunctionSubmitted.apply(submittedVariant1), submittedVariant1, 1);
+        DbsnpVariantsWrapper wrapper1 = buildSimpleWrapper(Collections.singletonList(submittedVariantEntity1));
 
         SubmittedVariant submittedVariant2 = new SubmittedVariant("assembly", TAXONOMY_1, "project", "contig", START,
                                                                   "reference", "alternate", CLUSTERED_VARIANT_ACCESSION,
                                                                   DEFAULT_SUPPORTED_BY_EVIDENCE, DEFAULT_ASSEMBLY_MATCH,
                                                                   DEFAULT_ALLELES_MATCH, DEFAULT_VALIDATED);
         DbsnpSubmittedVariantEntity submittedVariantEntity2 = new DbsnpSubmittedVariantEntity(
                 SUBMITTED_VARIANT_ACCESSION, hashingFunctionSubmitted.apply(submittedVariant2), submittedVariant2, 1);
-        DbsnpVariantsWrapper secondWrapper = buildSimpleWrapper(Collections.singletonList(submittedVariantEntity2));
+        DbsnpVariantsWrapper wrapper2 = buildSimpleWrapper(Collections.singletonList(submittedVariantEntity2));
+
+        DbsnpSubmittedVariantOperationEntity operationEntity1 = createOperation(submittedVariant1);
+        wrapper1.setOperations(Collections.singletonList(operationEntity1));
+        DbsnpSubmittedVariantOperationEntity operationEntity2 = createOperation(submittedVariant2);
+        wrapper2.setOperations(Collections.singletonList(operationEntity2));
+
+        dbsnpVariantsWriter.write(Arrays.asList(wrapper1, wrapper2));
+        assertClusteredVariantStored(wrapper1);
+        assertClusteredVariantDeclusteredStored(wrapper1);
+    }
 
-        dbsnpVariantsWriter.write(Arrays.asList(firstWrapper, secondWrapper));
-        assertClusteredVariantStored(firstWrapper);
+    private void assertClusteredVariantDeclusteredStored(DbsnpVariantsWrapper wrapper) {
```

Comment:
what is the difference between this method and this other one?
assertClusterVariantDeclusteredStored

Reply:
It was the same, deleted this method